CurrencyConverterControl/Themes/Generic.xaml

```diff
@@ -0,0 +1,80 @@
+                                    RelativeSource={RelativeSource AncestorType={x:Type local:CurrencyConverterControl}}}" />
+                        </Style>
+                        <local:ComboboxValueConverter  x:Key="ValueConverter"/>
+                    </ControlTemplate.Resources>
+                    <Grid Background="{TemplateBinding Background}" >
+                        <Grid.Resources>
+                            <DataTemplate x:Key="currenciesList">
+                                <StackPanel Orientation="Horizontal">
+                                    <Image  Height="10"  Source="{Binding Path=CurrencyFlag}"/>
+                                    <TextBlock Text="{Binding Path=CurrencyCode}"/>
+                                </StackPanel>                             
+                            </DataTemplate>                            
+                        </Grid.Resources>
+                        <Grid.ColumnDefinitions>
+                            <ColumnDefinition Width="*"/>
+                            <ColumnDefinition Width="*"/>
+                            <ColumnDefinition Width="*"/>
+                            <ColumnDefinition Width="*"/>
+                        </Grid.ColumnDefinitions>
+                       
+                        
+                        <!--Input Value to convert-->
+                        <TextBox  x:Name="TxtInput" Grid.Column="0" Text="{Binding InputValue, RelativeSource={RelativeSource AncestorType={x:Type local:CurrencyConverterControl}},
+                            UpdateSourceTrigger=PropertyChanged , StringFormat=N2}" 
+                            HorizontalAlignment="Left" Width="130" Height="22"/>
+                        <ComboBox x:Name="CmbSource"  SelectedIndex="0"
+                            ItemsSource="{Binding RelativeSource={RelativeSource AncestorType={x:Type local:CurrencyConverterControl}},
+                            Path=CurrencyList}"
+                                  SelectedValue="{Binding SourceCurrency, RelativeSource={RelativeSource
+                                                AncestorType={x:Type local:CurrencyConverterControl}}, Converter={StaticResource ValueConverter}, 
+                                                    Mode=TwoWay}"                                    
+                                            SelectedValuePath="CurrencyCode"
+                                  
+                                  Grid.Column="1" Height="22" Width="120" Background="Transparent"                                   
+                                  ItemTemplate="{StaticResource currenciesList}">
+                        </ComboBox>
+                        <!--Output value to convert-->
+                        <TextBox  x:Name="txtOutput" Grid.Column="2" Text="{Binding OutputValue,
+                                  RelativeSource={RelativeSource AncestorType={x:Type local:CurrencyConverterControl}},
```

Comment:
please check whether you can use {RelativeSource TemplatedParent} here. If I'm not wrong, it is get evaluated at compile time and only searchies up to templated parent control. While RelativeSource to some AncestorType is evaluated at runtime and can walk up the visual tree out of control template. It might hurt application performance.